## Nama : Alfarabi

### Objektif: Melakukan pengumpulan data penjualan produk seblak dari platform e-commerce Tokopedia menggunakan teknik web scraping, lalu membuat data preparation untuk pengolahan data dan melakukan analisis data yang sudah diambil.

In [3]:
! pip install bs4 selenium

## Library

In [1]:
# Library yang digunakan
import pandas as pd
from bs4 import BeautifulSoup
from selenium import webdriver
import time
from scipy import stats
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

## A. Web Scraping

### 1. Pengambilan data dari halaman pencarian kata kunci produk "seblak".

In [1]:
# URL yang akan diakses
url = 'https://www.tokopedia.com/search?navsource=&page=1&q=seblak&srp_component_id=02.01.00.00&srp_page_id=&srp_page_title=&st='
driver = webdriver.Chrome()
driver.get(url)
time.sleep(0.5)

# Membuat objek BeautifulSoup untuk parsing HTML halaman web
soup = BeautifulSoup(driver.page_source, 'html.parser')

NameError: name 'webdriver' is not defined

In [ ]:
soup.find_all('div',{"class":"prd_link-product-name css-3um8ox"})

In [ ]:
for div_name in soup.find_all('div',{"class":"prd_link-product-name css-3um8ox"}):
    print(div_tag.get_text())

### 2. Pengambilan data Nama Produk, Harga Produk, Penjual, Kota Toko, Banyaknya Terjual, dan Rating Produk.

In [ ]:
# Pembuatan WebDriver
driver = webdriver.Chrome()

# Pembuatan list
data = []

# Loop untuk menelusuri halaman pencarian Tokopedia
for i in range(1, 11):
    url = "https://www.tokopedia.com/search?navsource=&page={}&q=seblak&srp_component_id=02.01.00.00&srp_page_id=&srp_page_title=&st=".format(i)
    driver.get(url)
    time.sleep(1)
    soup = BeautifulSoup(driver.page_source, "html.parser")

    # Loop melalui setiap barang pada halaman
    for Produk in soup.find_all('div', {"class": "css-1asz3by"}):
        try:
            Nama = barang.find('div', {"class": "prd_link-product-name css-3um8ox"}).text
        except AttributeError:
            Nama = None

        try:
            Harga = barang.find('div', {"class": "prd_link-product-price css-h66vau"}).text
        except AttributeError:
            Harga = None

        try:
            Penjual = barang.find('span', {"class": "prd_link-shop-name css-1kdc32b flip"}).text
        except AttributeError:
            Penjual = None

        try:
            Lokasi = barang.find('span', {"class": "prd_link-shop-loc css-1kdc32b flip"}).text
        except AttributeError:
            Lokasi = None

        try:
            Terjual = barang.find('span', {"class": "prd_label-integrity css-1sgek4h"}).text
        except AttributeError:
            Terjual = None

        try:
            Rating = barang.find('span', {"class": "prd_rating-average-text css-t70v7i"}).text
        except AttributeError:
            Rating = None

        # Menambahkan data ke dalam list
        data.append((Nama, Harga, Penjual, Lokasi, Terjual, Rating))

# Untuk menutup WebDriver agar tidak running terus
driver.close()

# Membuat DataFrame dari list data
df = pd.DataFrame(data, columns=['Nama_Produk', 'Harga_Produk', 'Penjual', 'Kota_Toko', 'Banyaknya_Terjual', 'Rating_Produk'])
df

## B. Data Preparation

### 1. Data Exploration.

In [4]:
# Untuk membaca data dari file CSV 
df = pd.read_csv('P0G3_Alfarabi')
df

Nama_Produk Harga_Produk  \
0                              Kylafood Seblak karuhun     Rp11.880   
1                   Kylafood Seblak Mie Baso Sapi Play     Rp24.200   
2    Kylafood Mix Fav Selebgram Seblak Original & B...     Rp44.340   
3                             Kylafood Seblak Original     Rp15.750   
4    Geli Food Seblak Ceker Tanpa Tulang Super Peda...     Rp24.000   
..                                                 ...          ...   
362                      [250 gr] Kerupuk Seblak Pedas     Rp11.500   
363  Gelifood Seblak Instan Kerupuk Mawar Bumbu Ken...     Rp15.000   
364  Geli Food Seblak Ceker Tanpa Tulang Super Peda...     Rp24.000   
365           Seblak Instan Komplit Pedas Gurih Nikmat      Rp5.999   
366                      Seblak Instan Pedas Home Made      Rp3.500   

            Penjual          Kota_Toko Banyaknya_Terjual  Rating_Produk  
0               NaN                NaN     4 rb+ terjual            4.8  
1               NaN                NaN      250+ terjual            4.9  
2               NaN                NaN      100+ terjual            4.8  
3          kylafood            Bandung     10rb+ terjual            4.9  
4          Lidigeli         Kab. Garut      100+ terjual            4.8  
..              ...                ...               ...            ...  
362     CamilanJati      Kab. Sidoarjo       50+ terjual            5.0  
363        Lidigeli         Kab. Garut      250+ terjual            4.8  
364        Lidigeli         Kab. Garut      100+ terjual            4.8  
365  Baso Aci Ayang         Kab. Garut      3rb+ terjual            4.9  
366      the Dhecip  Tangerang Selatan      3rb+ terjual            4.9  

[367 rows x 6 columns]

- Langkah pertama saya akan menapilkan informasi dasar dari hasil data webscraping dengan menggunakan `.info()`.

In [5]:
# Untuk menampilkan informasi DataFrame
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 367 entries, 0 to 366
Data columns (total 6 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Nama_Produk        367 non-null    object 
 1   Harga_Produk       367 non-null    object 
 2   Penjual            350 non-null    object 
 3   Kota_Toko          350 non-null    object 
 4   Banyaknya_Terjual  367 non-null    object 
 5   Rating_Produk      357 non-null    float64
dtypes: float64(1), object(5)
memory usage: 17.3+ KB


- Lalu saya akan menampilkan beberapa baris dari data dengan menggunakan `.head()`.

In [6]:
# Melihat 5 baris teratas
df.head()

Nama_Produk Harga_Produk   Penjual  \
0                            Kylafood Seblak karuhun     Rp11.880       NaN   
1                 Kylafood Seblak Mie Baso Sapi Play     Rp24.200       NaN   
2  Kylafood Mix Fav Selebgram Seblak Original & B...     Rp44.340       NaN   
3                           Kylafood Seblak Original     Rp15.750  kylafood   
4  Geli Food Seblak Ceker Tanpa Tulang Super Peda...     Rp24.000  Lidigeli   

    Kota_Toko Banyaknya_Terjual  Rating_Produk  
0         NaN     4 rb+ terjual            4.8  
1         NaN      250+ terjual            4.9  
2         NaN      100+ terjual            4.8  
3     Bandung     10rb+ terjual            4.9  
4  Kab. Garut      100+ terjual            4.8

- Setelah itu saya menampilkan informasi dari nama kolom

In [7]:
# Melihat nama kolom
df.columns

Index(['Nama_Produk', 'Harga_Produk', 'Penjual', 'Kota_Toko',
       'Banyaknya_Terjual', 'Rating_Produk'],
      dtype='object')

- Lalu saya menampilkan statistik deskriptif dari DataFrame dengan menggunakan `.describe()`.

In [8]:
df.describe()

Rating_Produk
count     357.000000
mean        4.894678
std         0.101946
min         4.300000
25%         4.800000
50%         4.900000
75%         5.000000
max         5.000000

- Setelah itu saya menampilkan jumlah nilai null dalam setiap kolom DataFrame dengan menggunakan `.isnull().sum()`.

In [124]:
# Mengecek missing value
df.isnull().sum()

Nama_Produk           0
Harga_Produk          0
Penjual              17
Kota_Toko            17
Banyaknya_Terjual     0
Rating_Produk        10
dtype: int64

**Insight:**

- Pemahaman tentang distribusi variabel-variabel dalam dataset.
- Melihat adanya missing value dalam kolom DataFrame.

### 2. Data cleaning

In [125]:
# Mengecek unique value
df.Harga_Produk.unique()

array(['Rp11.880', 'Rp24.200', 'Rp44.340', 'Rp15.750', 'Rp24.000',
       'Rp26.600', 'Rp7.138', 'Rp13.900', 'Rp23.750', 'Rp4.450',
       'Rp3.500', 'Rp13.000', 'Rp10.000', 'Rp9.900', 'Rp25.000',
       'Rp9.999', 'Rp16.000', 'Rp13.999', 'Rp39.500', 'Rp20.000',
       'Rp15.000', 'Rp11.500', 'Rp5.999', 'Rp34.980', 'Rp28.500',
       'Rp9.000', 'Rp14.000', 'Rp22.000', 'Rp3.800', 'Rp55.000',
       'Rp10.600', 'Rp17.500', 'Rp22.688', 'Rp18.900', 'Rp24.225',
       'Rp5.500', 'Rp6.999', 'Rp9.916', 'Rp12.900', 'Rp20.482',
       'Rp35.625', 'Rp12.500', 'Rp32.000', 'Rp4.951', 'Rp28.000',
       'Rp4.500', 'Rp5.799', 'Rp7.500', 'Rp10.200', 'Rp26.040',
       'Rp16.500', 'Rp2.600', 'Rp2.500', 'Rp18.000', 'Rp35.999',
       'Rp26.000', 'Rp19.399', 'Rp39.900', 'Rp29.000', 'Rp64.000',
       'Rp49.680', 'Rp40.999', 'Rp25.500', 'Rp8.500', 'Rp12.999',
       'Rp18.500', 'Rp25.200', 'Rp38.000', 'Rp7.599', 'Rp3.000',
       'Rp36.900', 'Rp6.000', 'Rp9.950', 'Rp28.210', 'Rp9.800',
       'Rp11.900',

In [128]:
# Mengganti karakter yang tidak diperlukan
df['Harga_Produk'] = df['Harga_Produk'].str.replace('Rp', '').str.replace('.', '')

In [55]:
# Mengecek unique value
df.Banyaknya_Terjual.unique()

array(['4 rb+ terjual', '250+ terjual', '100+ terjual', '10rb+ terjual',
       '60+ terjual', '7rb+ terjual', '1rb+ terjual', '500+ terjual',
       '3rb+ terjual', '70+ terjual', '50+ terjual', '30+ terjual',
       '40+ terjual', '80+ terjual', '21 terjual', '9 terjual',
       '24 terjual', '28 terjual', '2 terjual', '5 terjual',
       '90+ terjual', '8 terjual', '6 terjual', '10 terjual',
       '3 rb+ terjual', '2 rb+ terjual', '4rb+ terjual', '750+ terjual',
       '1 terjual', '7 terjual', '2rb+ terjual', '26 terjual',
       '11 terjual', '3 terjual'], dtype=object)

In [134]:
# Mengganti karakter yang tidak diperlukan
df['Banyaknya_Terjual'] = df['Banyaknya_Terjual'].str.replace('terjual', '').str.replace(' ', '').str.replace('\+', '').str.replace('+', '').str.replace('rb', '000')
df['Banyaknya_Terjual'].fillna(0, inplace=True)

In [135]:
# Mengganti nilai NaN di kolom Rating_Produk dengan 0
df['Rating_Produk'].fillna(0, inplace=True)

In [137]:
# Menyesuaikan tipe data kolom
df['Harga_Produk'] = df['Harga_Produk'].astype(int)
df['Banyaknya_Terjual'] = df['Banyaknya_Terjual'].astype(int)
df['Rating_Produk'] = df['Rating_Produk'].astype(float)

In [138]:
# Melihat tipe data apakah sudah sesuai atau belum
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 367 entries, 0 to 366
Data columns (total 6 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Nama_Produk        367 non-null    object 
 1   Harga_Produk       367 non-null    int32  
 2   Penjual            350 non-null    object 
 3   Kota_Toko          350 non-null    object 
 4   Banyaknya_Terjual  367 non-null    int32  
 5   Rating_Produk      367 non-null    float64
dtypes: float64(1), int32(2), object(3)
memory usage: 14.5+ KB


In [139]:

# Menyimpan file yang udah di cleaning ke dalam format CSV
df.to_csv('P0G3_Alfarabi.csv',index=False)

**Insight:**

- Mengidentifikasi kolom atau variabel yang memiliki nilai kosong atau hilang.
- Menampilkan unique value
- Melakukan normalisasi data
- Memperbaiki Kesalahan Tipe Data

## C. Business Understanding/Problem Statement

### SMART

##### **Specific**: Meningkatkan penjualan seblak melalui platform e-commerce dengan fokus pada peningkatan visibilitas produk.

##### **Measurable**: Menetapkan target peningkatan penjualan seblak sebesar 20% dalam enam bulan pertama setelah implementasi rencana.

##### **Achievable**: Memastikan ketersediaan stok yang cukup untuk memenuhi permintaan yang diperkirakan meningkat.

##### **Relevant**: Memperluas jangkauan pasar melalui platform e-commerce untuk mencapai pelanggan potensial yang lebih luas, termasuk di daerah-daerah di luar kota tempat penjualan fisik berlokasi.

##### **Time-Bound**: Memantau dan mengevaluasi kinerja penjualan setiap bulan untuk memastikan bahwa target peningkatan 20% tercapai dalam enam bulan.

## D. Analysis

In [140]:
import csv

In [3]:
# menampilkan hasil data yang udah di cleaning
df_1 = pd.read_csv('P0G3_Alfarabi.csv')
df_1

Nama_Produk  Harga_Produk  \
0                              Kylafood Seblak karuhun         11880   
1                   Kylafood Seblak Mie Baso Sapi Play         24200   
2    Kylafood Mix Fav Selebgram Seblak Original & B...         44340   
3                             Kylafood Seblak Original         15750   
4    Geli Food Seblak Ceker Tanpa Tulang Super Peda...         24000   
..                                                 ...           ...   
362                      [250 gr] Kerupuk Seblak Pedas         11500   
363  Gelifood Seblak Instan Kerupuk Mawar Bumbu Ken...         15000   
364  Geli Food Seblak Ceker Tanpa Tulang Super Peda...         24000   
365           Seblak Instan Komplit Pedas Gurih Nikmat          5999   
366                      Seblak Instan Pedas Home Made          3500   

            Penjual          Kota_Toko  Banyaknya_Terjual  Rating_Produk  
0               NaN                NaN               4000            4.8  
1               NaN                NaN                250            4.9  
2               NaN                NaN                100            4.8  
3          kylafood            Bandung              10000            4.9  
4          Lidigeli         Kab. Garut                100            4.8  
..              ...                ...                ...            ...  
362     CamilanJati      Kab. Sidoarjo                 50            5.0  
363        Lidigeli         Kab. Garut                250            4.8  
364        Lidigeli         Kab. Garut                100            4.8  
365  Baso Aci Ayang         Kab. Garut               3000            4.9  
366      the Dhecip  Tangerang Selatan               3000            4.9  

[367 rows x 6 columns]

In [42]:
df_1.describe()

Harga_Produk  Banyaknya_Terjual  Rating_Produk
count    367.000000         367.000000     367.000000
mean   16647.743869        1244.348774       4.761308
std    10463.981137        2548.168131       0.804276
min     2500.000000           1.000000       0.000000
25%     9900.000000          50.000000       4.800000
50%    15000.000000         100.000000       4.900000
75%    24000.000000         500.000000       5.000000
max    64000.000000       10000.000000       5.000000

### 1. Mencari rata-rata, median, standar deviasi, skewness, dan kurtosis.

- Saya mencari rata-rata dengan menggunakan fungsi `.mean()`.

In [8]:
# Mencari nilai mean
print(f'Nilai Rata-Rata Harga Produk adalah : {df_1.Harga_Produk.mean()}\n')
print(f'Nilai Rata-Rata Banyaknya Terjual adalah : {df_1.Harga_Produk.mean()}\n')
print(f'Nilai Rata-Rata Rating Produk adalah : {df_1.Harga_Produk.mean()}\n')

Nilai Rata-Rata Harga Produk adalah : 16647.743869209808

Nilai Rata-Rata Banyaknya Terjual adalah : 16647.743869209808

Nilai Rata-Rata Rating Produk adalah : 16647.743869209808



- Lalu saya mencari nilai median dengan menggunakan fungsi `.median()`.

In [15]:
# Mencari nilai median
print(f'Nilai Median Harga Produk adalah : {df_1.Harga_Produk.median()}\n')
print(f'Nilai Median Banyaknya Terjual adalah : {df_1.Harga_Produk.median()}\n')
print(f'Nilai Median Rating Produk adalah : {df_1.Harga_Produk.median()}\n')

Nilai Median Harga Produk adalah : 15000.0

Nilai Median Banyaknya Terjual adalah : 15000.0

Nilai Median Rating Produk adalah : 15000.0



- Selanjutnya saya mencari nilai standar deviasi menggunakan fungsi `.std()`.

In [14]:
# Mencari nilai standar deviasi
print(f'Nilai Standar Deviasi Harga Produk adalah : {df_1.Harga_Produk.std()}\n')
print(f'Nilai Standar Deviasi Banyaknya Terjual adalah : {df_1.Harga_Produk.std()}\n')
print(f'Nilai Standar Deviasi Produk adalah : {df_1.Harga_Produk.std()}\n')

Nilai Standar Deviasi Harga Produk adalah : 10463.981136704117

Nilai Standar Deviasi Banyaknya Terjual adalah : 10463.981136704117

Nilai Standar Deviasi Produk adalah : 10463.981136704117



- Setelah itu saya mencari nilai skew dengan mengunakan fungsi `.skew()`.

In [11]:
# Mencari nilai skewness
print(f'Nilai Skewness Harga Produk adalah : {df_1.Harga_Produk.skew()}\n')
print(f'Nilai Skewness Banyaknya Terjual adalah : {df_1.Harga_Produk.skew()}\n')
print(f'Nilai Skewness Rating Produk adalah : {df_1.Harga_Produk.skew()}\n')

Nilai Skewness Harga Produk adalah : 0.9599332642271041

Nilai Skewness Banyaknya Terjual adalah : 0.9599332642271041

Nilai Skewness Rating Produk adalah : 0.9599332642271041



- Kemudian saya mencari nilai kurtosis menggunakan fungsi `.kurtosis()`.

In [12]:
# Mencari nilai kurtosis
print(f'Nilai Kurtosis Harga Produk adalah : {df_1.Harga_Produk.kurtosis()}\n')
print(f'Nilai Kurtosis Banyaknya Terjual adalah : {df_1.Harga_Produk.kurtosis()}\n')
print(f'Nilai Kurtosis Rating Produk adalah : {df_1.Harga_Produk.kurtosis()}\n')

Nilai Kurtosis Harga Produk adalah : 1.3596747144281944

Nilai Kurtosis Banyaknya Terjual adalah : 1.3596747144281944

Nilai Kurtosis Rating Produk adalah : 1.3596747144281944



In [13]:
df_1.describe()

Harga_Produk  Banyaknya_Terjual  Rating_Produk
count    367.000000         367.000000     367.000000
mean   16647.743869        1244.348774       4.761308
std    10463.981137        2548.168131       0.804276
min     2500.000000           1.000000       0.000000
25%     9900.000000          50.000000       4.800000
50%    15000.000000         100.000000       4.900000
75%    24000.000000         500.000000       5.000000
max    64000.000000       10000.000000       5.000000

Setelah itu, saya akan melakukan pengelompokan barang yang telah terjual berdasarkan lokasi kota.

In [56]:
barang_terjual = df_1[['Kota_Toko','Banyaknya_Terjual']].groupby('Kota_Toko').sum()
barang_terjual.head()

Banyaknya_Terjual
Kota_Toko                    
Bandung                 20345
Banjarbaru                 11
Bekasi                     63
Cimahi                  21000
Depok                   17640

**Insight:**

dari data diatas kita dapat melihat:
- Pusat distribusi data (rata-rata dan median).
- Variabilitas atau dispersi data (standar deviasi).
- Asimetri distribusi data (skewness).
- Keruncingan distribusi data (kurtosis).

### 2. Confidence Interval

- Lalu saya mencari nilai lower value dan upper value.

In [40]:
std = barang_terjual['Banyaknya_Terjual'].std()
N = len(barang_terjual)
low, up = stats.norm.interval(0.95,loc=barang_terjual['Banyaknya_Terjual'].mean(),scale=std/np.sqrt(N))
print('Lower Value:',low)
print('Upper Value:',up)

Lower Value: -809.421371249191
Upper Value: 37015.75470458253


### 3. Hypotesist Testing

In [51]:
# Untuk melihat unique value dari Kota Toko
df_1.Kota_Toko.unique()

array([nan, 'Bandung', 'Kab. Garut', 'Jakarta Selatan', 'Cimahi',
       'Kab. Tangerang', 'Depok', 'Jakarta Timur', 'Tangerang Selatan',
       'Surabaya', 'Jakarta Barat', 'Kab. Bogor', 'Surakarta',
       'Kab. Bandung', 'Kab. Sidoarjo', 'Bekasi', 'Semarang', 'Tangerang',
       'Kab. Majalengka', 'Jakarta Pusat', 'Tasikmalaya', 'Kab. Jombang',
       'Banjarbaru', 'Serang', 'Kab. Malang'], dtype=object)

- Setelah itu saya akan melakukan pengelompokan barang yang telah terjual berdasarkan lokasi kota.

In [52]:
jabodetabek = df_1[(df_1["Kota_Toko"]=="Tangerang") | (df_1["Kota_Toko"]=="Tangerang Selatan") | (df_1["Kota_Toko"]=="Jakarta Barat") | (df_1["Kota_Toko"]=="Jakarta Selatan") | (df_1["Kota_Toko"]=="Jakarta Barat") | (df_1["Kota_Toko"]=="Jakarta Timur") | (df_1["Kota_Toko"]=="Jakarta Pusat") | (df_1["Kota_Toko"]=="Kab. Tangerang") | (df_1["Kota_Toko"]=="Kab. Bogor") | (df_1["Kota_Toko"]=="Bekasi")]

jabodetabek

Nama_Produk  Harga_Produk  \
5             Seblak Rafael/Seblak Coet Instan Halal         26600   
7                                SEBLAK INSTAN MOMMY         13900   
9                            SEBLAK INSTANT TERMURAH          4450   
10                     Seblak Instan Pedas Home Made          3500   
12                   Kylafood Seblak Rempah Autentik         10000   
..                                               ...           ...   
355  Kerupuk SEBLAK MAWAR Rafael - Cikruh Daun Jeruk         20000   
358           Seblak Rafael/Seblak Coet Instan Halal         26600   
359                          SEBLAK INSTANT TERMURAH          4450   
360                    Seblak Instan Pedas Home Made          3500   
366                    Seblak Instan Pedas Home Made          3500   

                            Penjual          Kota_Toko  Banyaknya_Terjual  \
5                        Foodstocks    Jakarta Selatan                 60   
7                  Rav Jaya Display     Kab. Tangerang               1000   
9    Pusat Cuankie & Cemilan Frozen      Jakarta Timur              10000   
10                       the Dhecip  Tangerang Selatan               3000   
12                 Kylafood Jakarta    Jakarta Selatan                100   
..                              ...                ...                ...   
355                      KRIUK JAYA      Jakarta Barat                 40   
358                      Foodstocks    Jakarta Selatan                 60   
359  Pusat Cuankie & Cemilan Frozen      Jakarta Timur              10000   
360                      the Dhecip  Tangerang Selatan               3000   
366                      the Dhecip  Tangerang Selatan               3000   

     Rating_Produk  
5              5.0  
7              4.8  
9              4.8  
10             4.9  
12             5.0  
..             ...  
355            4.9  
358            5.0  
359            4.8  
360            4.9  
366            4.9  

[188 rows x 6 columns]

In [53]:
non_jabodetabek = df_1[~(df_1["Kota_Toko"]=="Tangerang") | (df_1["Kota_Toko"]=="Tangerang Selatan") | (df_1["Kota_Toko"]=="Jakarta Barat") | (df_1["Kota_Toko"]=="Jakarta Selatan") | (df_1["Kota_Toko"]=="Jakarta Barat") | (df_1["Kota_Toko"]=="Jakarta Timur") | (df_1["Kota_Toko"]=="Jakarta Pusat") | (df_1["Kota_Toko"]=="Kab. Tangerang") | (df_1["Kota_Toko"]=="Kab. Bogor") | (df_1["Kota_Toko"]=="Bekasi")]

non_jabodetabek

Nama_Produk  Harga_Produk  \
0                              Kylafood Seblak karuhun         11880   
1                   Kylafood Seblak Mie Baso Sapi Play         24200   
2    Kylafood Mix Fav Selebgram Seblak Original & B...         44340   
3                             Kylafood Seblak Original         15750   
4    Geli Food Seblak Ceker Tanpa Tulang Super Peda...         24000   
..                                                 ...           ...   
362                      [250 gr] Kerupuk Seblak Pedas         11500   
363  Gelifood Seblak Instan Kerupuk Mawar Bumbu Ken...         15000   
364  Geli Food Seblak Ceker Tanpa Tulang Super Peda...         24000   
365           Seblak Instan Komplit Pedas Gurih Nikmat          5999   
366                      Seblak Instan Pedas Home Made          3500   

            Penjual          Kota_Toko  Banyaknya_Terjual  Rating_Produk  
0               NaN                NaN               4000            4.8  
1               NaN                NaN                250            4.9  
2               NaN                NaN                100            4.8  
3          kylafood            Bandung              10000            4.9  
4          Lidigeli         Kab. Garut                100            4.8  
..              ...                ...                ...            ...  
362     CamilanJati      Kab. Sidoarjo                 50            5.0  
363        Lidigeli         Kab. Garut                250            4.8  
364        Lidigeli         Kab. Garut                100            4.8  
365  Baso Aci Ayang         Kab. Garut               3000            4.9  
366      the Dhecip  Tangerang Selatan               3000            4.9  

[363 rows x 6 columns]

- Saya akan melakukan Hypothesis Test dengan menggunakan `Two-Sample Independent Test`.

In [54]:
area_jabodetabek = ['Tangerang', 'Tangerang Selatan', 'Jakarta Barat', 'Jakarta Selatan', 'Jakarta Barat', 'Jakarta Timur', 'Jakarta Pusat', 'Kab. Tangerang', 'Kab. Bogor', 'Bekasi']

hargaJabodetakbek = df_1[df_1['Kota_Toko'].isin(area_jabodetabek)]
meanHargaJabodetabek = hargaJabodetakbek['Harga_Produk'].mean()

hargaNonJabodetabek = df_1[~df_1['Kota_Toko'].isin(area_jabodetabek)]
meanHargaNonJabodetabek = hargaNonJabodetabek['Harga_Produk'].mean()

print(f"Mean Harga Jabodetabek: Rp. {meanHargaJabodetabek}")
print(f"Mean Harga non Jabodetabek: Rp. {meanHargaNonJabodetabek}")

Mean Harga Jabodetabek: Rp. 16751.11170212766
Mean Harga non Jabodetabek: Rp. 16539.178770949722


In [187]:
t_stat, p_val = stats.ttest_ind(hargaJabodetakbek['Harga_Produk'], hargaNonJabodetabek['Harga_Produk'])

print('T-Statistic:', t_stat)
print('P_Value:', p_val)

T-Statistic: 0.1936877310222945
P_Value: 0.8465281409209847


**Insight:**

berdasarkan data diatass dapat disimpulkan:
- H0: Harga barang di jabodetabek == harga barang diluar jabodetabek
- H1: Harga barang di jabodetabek != harga barang diluar jabodetabek

### 4. Correlation Test

In [4]:
# Menggunakan spearmanr untuk menghitung korelasi dikarenakan data tidak terdistribusi normal
corr_rho, pval_s = stats.spearmanr(df_1['Harga_Produk'], df_1['Banyaknya_Terjual'])

print(f"rho-correlation: {corr_rho:.2f}, p-value: {pval_s}")

rho-correlation: -0.57, p-value: 7.028600209953802e-33


**Insight:**

Berdasarkan hasil perhitungan, hubungan antara harga produk dan jumlah penjualan dianggap nyata karena nilai p-value kurang dari 0,05.

## E. Conclusion

1. - Nilai Rata-Rata Harga Produk adalah : 16647.743869209808 dan Nilai Median Harga Produk adalah : 15000.0
   - Nilai Rata-Rata Banyaknya Terjual adalah : 16647.743869209808 dan Nilai Median Banyaknya Terjual adalah : 15000.0
   - Nilai Rata-Rata Rating Produk adalah : 16647.743869209808 dan Nilai Median Rating Produk adalah : 15000.0
   dan nilai tidak terdistribusi dengan normal karena skewness >1

2. Batas minimum barang terjual: -809.421371249191 dan Batas maximal barang terjual: 37015.75470458253

3. Harga barang di jabodetabek tidak sama dengan harga barang diluar jabodetabek

4. Ada hubungan antara harga produk dengan banyaknya terjual
